auction-server/src/auction/service/workers.rs

```diff
@@ -147,21 +151,22 @@ impl Service<Svm> {
                                 log = ?rpc_log.clone(),
                                 "New log trigger received",
                             );
+                            if let Ok(signature) = Signature::from_str(&rpc_log.value.signature){
                             self.task_tracker.spawn({
                                 let service = self.clone();
                                 async move {
                                     let submitted_auctions = service.repo.get_in_memory_submitted_auctions().await;
                                     if let Some(auction) = submitted_auctions.iter().find(|auction| {
-                                        auction.tx_hash.map(|tx_hash|
-                                            tx_hash.to_string() == rpc_log.value.signature,
-                                        ).unwrap_or(false)
+                                        auction.bids.iter().any(|bid| bid.chain_data.transaction.signatures[0] == signature)
                                     }) {
+                                        service.config.chain_config.simulator.remove_pending_transaction(&signature).await;
```

Comment:
should this be part of conclude_auction?

Reply:
moved it to get_bid_status, marginally better than here but still weird.

Not easy to put it inside conclude_auction since it's generic.